src/sonic-config-engine/tests/multi_npu_data/sample_port_config-0.ini

```diff
@@ -1,9 +1,9 @@
-# name          lanes          alias            asic_port_name
-Ethernet0     33,34,35,36      Ethernet1/1      Eth0-ASIC0
-Ethernet4     29,30,31,32      Ethernet1/2      Eth1-ASIC0
-Ethernet8     41,42,43,44      Ethernet1/3      Eth2-ASIC0
-Ethernet12    37,38,39,40      Ethernet1/4      Eth3-ASIC0
-Ethernet-BP0  13,14,15,16      Ethernet-BP0     Eth4-ASIC0
-Ethernet-BP4  17,18,19,20      Ethernet-BP4     Eth5-ASIC0
-Ethernet-BP8  21,22,23,24      Ethernet-BP8     Eth6-ASIC0
-Ethernet-BP12 25,26,27,28      Ethernet-BP12    Eth7-ASIC0
\ No newline at end of file
+# name          lanes          alias           asic_port_name    role 
+Ethernet0     33,34,35,36      Ethernet1/1      Eth0-ASIC0         E
```

Comment:
I talked to Joe, as per him it would be good to add the "index" field also in port_config.ini files. There could be API's which use that.

Reply:
What is the usage or significance of index here? There is a way an index is computed based on interface name, which is used in SNMP mainly. 
https://github.com/Azure/sonic-py-swsssdk/blob/master/src/swsssdk/port_util.py#L26
Is the interface index compute in port_util same as index here?

Reply:
@judyjoseph, @SuvarnaMeenakshi. 
I have added port_index in the port_config.ini in the latest commit. The port index is a running number, there is no significance to it.

Reply:
Thanks !  To mention the significance of "index" .. there are some places where the code check for presence of "index" in the port_config.ini eg: https://github.com/Azure/sonic-platform-common/blob/be1cc2475e65b764f7aab44785a779cc3d011ee9/sonic_platform_base/sonic_sfp/sfputilbase.py#L458

So it would be safer to have the index field.... as there are logic below which checks for len(line.split()) >= 4: etc. This will change for us as we add more columns now.

Also we would need to have separate index numbering for front-panel and back panel interfaces ... 

Ethernet0     33,34,35,36      Ethernet1/1        0       Eth0-ASIC0         E
Ethernet4     29,30,31,32      Ethernet1/2        1       Eth1-ASIC0         E  
...
Ethernet-BP0  13,14,15,16      Ethernet-BP0       0       Eth4-ASIC0         I 
Ethernet-BP4  17,18,19,20      Ethernet-BP4       1       Eth5-ASIC0         I 
....

Reply:
> Also we would need to have separate index numbering for front-panel and back panel interfaces ...

@judyjoseph, can you elaborate why we need  separate index numbering for front-panel and back panel interface ?

Reply:
In the platform code ( could be other places as well ) we are making a logical list of all the interfaces, and we are looking for only front panel interfaces which is significant. Currently when there is no "index" field in port_config.ini, the way it calculates the index is from the name "Ethernet0"(here it is 0/4 ==> 0), "Ethernet4" (here it is 4/4 ==> 1) etc. 

So It would be ok to start the index from 0 separately per **interface type** and **it will follow the interface naming/numbering convention also**. Do you think there could be an issue ?

Reply:
@judyjoseph, Changed the port_config.ini in the latest commit